src/backend/distributed/commands/table.c

```diff
@@ -701,13 +701,81 @@ PostprocessAlterTableSchemaStmt(Node *node, const char *queryString)
 }
 
 
+static char *
+GenerateIndexConstraintName(const char *tabname, Oid namespaceId, Constraint *constraint)
+{
+	char *conname = NULL;
+
```

Comment:
You are using ChooseIndexName in all the subcases of this function. Maybe here you can declare all the variables with meaningful names, e.g. something like:
bool primary
bool isconstraint
List *indexParams
List *excludeOpNames

And then set these values in each subcase: it also makes the difference in generating the name between the subcases clearer

Reply:
Looking at the postgres codebase, the general style looks like variables are not used for passing literal arguments to functions. We seem to adapt this coding practice in citus e.g.

https://github.com/citusdata/citus/blob/5268d0a6cbfc987a329f7fd77988428b585096aa/src/backend/distributed/commands/role.c#:~:text=const%20char%20*roleName%20%3D%20RoleSpecString(role%2C%20false)%3B
https://github.com/citusdata/citus/blob/5268d0a6cbfc987a329f7fd77988428b585096aa/src/backend/distributed/commands/alter_table.c#:~:text=CreateDistributedTable(con%2D%3EnewRelationId%2C%20NULL,NULL)%3B

There are counterwise examples too.  So it is hard to say it is a strict coding practice for citus. However, especially for this instance, the kind of verbosity which comes from assigning literals to variables within the function several times would cause code bloat without improving readability imo.

If you do not have strong opinion about this, I would prefer not to make this change.

Reply:
Makes sense. How about just the boolean values? I leave it up to you.